crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -503,9 +515,18 @@ impl<'a> Submitter<'a> {
     /// Estimate access list and validate
     async fn estimate_access_list(
         &self,
+        solver: &dyn Solver,
+        settlement: &Settlement,
         tx: &TransactionBuilder<Web3Transport>,
     ) -> Result<AccessList> {
-        let access_list = self.access_list_estimator.estimate_access_list(tx).await?;
+        let access_list = estimate_settlement_access_list(
+            self.access_list_estimator,
+            self.web3.clone(),
+            solver.account().clone(),
```

Comment:
I believe the only purpose to add `&dyn Solver` to many function signatures was to get it's `account` here but this is already stored in `Submitter::account`. Should slim down the PR by a lot to drop this.